### PR TITLE
(8255&8248): Updated gateway, spoke gateway and transit gateway resources

### PR DESCRIPTION
**Pull request by zhiqiangjin**
1. To support Default argument type
2. To allow refresh public_ip for gateway data source

### CODE REVIEW

File: aviatrix/resource_spoke_vpc.go [hunk 1]
```diff
@@ -323,16 +324,25 @@ func resourceAviatrixSpokeVpcUpdate(d *schema.ResourceData, meta interface{}) er
...
 				return fmt.Errorf("failed to enable single AZ GW HA: %s", err)
 			}
+		} else if singleAz == "disabled" {
+			log.Printf("[INFO] Enable Single AZ GW HA: %#v", singleAZGateway)
+			err := client.DisableSingleAZGateway(singleAZGateway)
+			if err != nil {
+				return fmt.Errorf("failed to enable single AZ GW HA: %s", err)
+			}
+		} else {
+			return fmt.Errorf("single_az_hs is not set correctly")
```
Comment (arvindsree) [inline aviatrix/resource_spoke_vpc.go]: Typo

Comment (arvindsree) [reply aviatrix/resource_spoke_vpc.go]: Also make it clear. Something like 'single_az_ha should be only "enabled"/"disabled"  ' is helpful

File: website/docs/d/aviatrix_gateway.html.markdown [hunk 2]
```diff
@@ -35,4 +35,8 @@ The following arguments are supported:
 * `vpc_id`       - AWS VPC ID.
 * `vpc_reg`      - AWS VPC Region. 
 * `vpc_size`     - Instance type.
+
+## The following arguments are computed - please do not edit in the resource file:
```
Comment (arvindsree) [inline website/docs/d/aviatrix_gateway.html.markdown]: The above arguments are also computed

Comment (zhiqiangjin) [reply website/docs/d/aviatrix_gateway.html.markdown]: yes

File: website/docs/d/aviatrix_gateway.html.markdown [hunk 3]
```diff
@@ -32,7 +32,12 @@ The following arguments are supported:
 * `cloud_type`   - Type of cloud service provider. (Only AWS is supported currently. Value of 1 for AWS.)
```
Comment (arvindsree) [inline website/docs/d/aviatrix_gateway.html.markdown]: cloud_type is set as Computed: true but read function is not setting it

**Pull request merged**